gems/pending/VMwareWebService/VimService.rb

```diff
@@ -892,8 +902,19 @@ def retrievePropertiesEx(propCol, specSet, max_objects = nil)
   end
 
   def retrievePropertiesCompat(propCol, specSet, max_objects = nil)
+    oc = VimArray.new('ArrayOfObjectContent')
+
     rv = retrievePropertiesEx(propCol, specSet, max_objects)
-    rv ? rv['objects'] : []
+    if rv
+      oc.concat(rv['objects'])
+
+      while rv && rv['token']
+        rv = continueRetrievePropertiesEx(propCol, rv['token'])
+        oc.concat(rv['objects']) unless rv.nil?
```

Comment:
@agrare - Doesn't this defeat the purpose? I thought the idea (at least in part) was to process smaller chunks of data, so we wouldn't have to load the whole inventory in memory at once. Here, we're loading it all into an array before processing it.